specs/eid-hermes-interface.md

```diff
@@ -0,0 +1,64 @@
+# eid-hermes Interface Specification
+
+## Introduction
+
+This document defines the PCIe device interface for the eid-hermes
+eBPF accelerator device.
+
+## Overall BAR Layout
+
+The eid-hermes PF BAR layout follows that of the shell of the generic
+[AWS F1 FPGA][1] with some minor modifications.
+
+```
+eid-hermes PF
+   |------- BAR0  
+   |         * 32-bit BAR, non-prefetchable
+   |         * 32MiB (0 to 0x1FF-FFFF)
+   |         * eid-hermes device registers  
+   |------- BAR2
+   |         * 64-bit BAR, prefetchable
+   |         * 64KiB (0 to 0xFFFF)
+   |         * eid-hermes XDMA block
+   |------- BAR4
+             * 64-bit BAR, prefetchable
+             * Size is TBD
+             * Maps the exposed memory on eid-hermes
+```
+## BAR0 Layout
+
+BAR0 in an eid-hermes device contains the eid-hermes device registers
+which advertise a range of relevant capabilities to the host. These
+registers are defined in the table below.
+
+|Offset  | Length | Name    | Mode | Value | Description            |
+|--------|--------|---------|------|-------|------------------------|
+| 0x00   | 1      | EHVER   | RO   | 1     | Interface version      |
+| 0x04   | 4      | EHTS    | RO   | NA    | Timestamp              |
+| 0x08   | 1      | EHENG   | RO   | NA    | Number of eBPF Engines |
+| 0x09   | 1      | EHSLOT  | RO   | NA    | Number of eBPF slots   |
+| 0x0A   | 1      | EHPM    | RO   | NA    | Number of eBPF program memories   |
```

Comment:
I don't think we need to limit the number of data slots to 1 byte, maybe let's make it 2 or 4 (program slots should be fine with 1, but it wouldn't hurt to increase it as well).

Reply:
I can't see us needing more than 255 program or data slots in eid-hermes.

Reply:
I guess that depends on the data slot size. And whether you want to preload everything or do it in batches.

Reply:
@iomartin I see data slots being quite large (16MiB?). I think we keep then as single byte for version 1.

Reply:
Even at 16MiB, with 256 data slots you can only store 4 GiB at a time. That's not really that much...